### ISSUE TITLE
Error in Unicode/ASCII conversion problem in BEL API

### ISSUE THREAD

**Issue by johnbachman** [sorgerlab/indra#29]
```

In [1]: from indra.bel import bel_api as ba
INFO:rdflib:RDFLib Version: 4.2.1

In [2]: bp = ba.process_ndex_neighborhood(['BRAF', 'MAP2K1'])
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): services.bigmech.ndexbio.org
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
Getting all direct statements...

Checking for 'degenerate' statements...

Protein -> Protein/Activity statements:
---------------------------------------

Total indirect statements: 166
Total direct statements: 40
Converted statements: 36
Degenerate statements: 0
>> Total unhandled statements: 4

--- Unhandled statements ---------
complex_p_HGNC_BRAF_p_HGNC_PRKCE_p_HGNC_RPS6KB2_DirectlyIncreases_kin_p_HGNC_PRKCE
p_HGNC_BRAF_sub_V_600_E_DirectlyIncreases_kin_p_HGNC_BRAF
a_CHEBI_sorafenib_tosylate_DirectlyDecreases_kin_p_HGNC_BRAF
complex_p_HGNC_BRAF_p_HGNC_RAF1_DirectlyIncreases_kin_p_HGNC_RAF1

--- Converted BelPy Statements -------------
0: Complex(['EIF4G1', 'EIF4E'], [])
1: Complex(['BRAF', 'RAF1'], [])
2: Complex(['MAP2K1', 'PTPN7'], [])
3: Complex(['RPS6KB2', 'PRKCE', 'BRAF'], [])
4: Complex(['MAP2K1', 'GIT1'], [])
5: Complex(['MAP2K1', 'RAF1'], [])
6: Phosphorylation(PRKA_FAMILY, BRAF, Phosphorylation, 429, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 11510412;10869359)])
7: Phosphorylation(AKT_FAMILY, BRAF, Phosphorylation, 429, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 11510412;10869359)])
8: Phosphorylation(MAP2K1, MAPK3, Phosphorylation, 202, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 8626767;93330262;15466476;11971971)])
9: Phosphorylation(RAF1, MAP2K1, Phosphorylation, 218, [Evidence(bel, http://bio2rdf.org/pubmed:12937126, [], MEK1/2 is activated through phosphorylation at Ser 217/Ser 221 catalyzed by Raf.)])
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-2-a69375dadcfa> in <module>()
----> 1 bp = ba.process_ndex_neighborhood(['BRAF', 'MAP2K1'])

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/bel_api.py in process_ndex_neighborhood(gene_names, rdf_out)
     13     with open(rdf_out, 'wt') as fh:
     14         fh.write(rdf.encode('utf-8'))
---> 15     bp = process_belrdf(rdf)
     16     bp.print_statements()
     17     return bp

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/bel_api.py in process_belrdf(rdf_str)
     35     bp.print_statement_coverage()
     36     print "\n--- Converted BelPy Statements -------------"
---> 37     bp.print_statements()
     38     return bp
     39 

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/processor.py in print_statements(self)
    662     def print_statements(self):
    663         for i, stmt in enumerate(self.statements):
--> 664             print "%s: %s" % (i, stmt)

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/statements.py in __str__(self)
    215         s = ("%s(%s, %s, %s, %s, %s)" %
    216                   (type(self).__name__, self.enz.name, self.sub.name, self.mod,
--> 217                    self.mod_pos, self.evidence))
    218         return s
    219 

UnicodeEncodeError: 'ascii' codec can't encode character u'\u2013' in position 127: ordinal not in range(128)
```

**Comment by bgyori** [comment]
Fixed in cc2ea04e41a365e36c802c64a3a731d996467968. Will be in master when preassembler is merged.

**Comment by johnbachman** [comment]
Here's another one:
```

In [1]: from indra.bel import bel_api as ba
INFO:rdflib:RDFLib Version: 4.2.1

In [2]: bp = ba.process_ndex_neighborhood(['BRAF', 'MAP2K1'])
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): services.bigmech.ndexbio.org
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
http://www.openbel.org/vocabulary/Phosphorylation
Getting all direct statements...

Checking for 'degenerate' statements...

Protein -> Protein/Activity statements:
---------------------------------------

Total indirect statements: 166
Total direct statements: 40
Converted statements: 36
Degenerate statements: 0
>> Total unhandled statements: 4

--- Unhandled statements ---------
complex_p_HGNC_BRAF_p_HGNC_RAF1_DirectlyIncreases_kin_p_HGNC_RAF1
p_HGNC_BRAF_sub_V_600_E_DirectlyIncreases_kin_p_HGNC_BRAF
complex_p_HGNC_BRAF_p_HGNC_PRKCE_p_HGNC_RPS6KB2_DirectlyIncreases_kin_p_HGNC_PRKCE
a_CHEBI_sorafenib_tosylate_DirectlyDecreases_kin_p_HGNC_BRAF

--- Converted BelPy Statements -------------
0: Complex(['EIF4G1', 'EIF4E'], [])
1: Complex(['RAF1', 'BRAF'], [])
2: Complex(['PTPN7', 'MAP2K1'], [])
3: Complex(['BRAF', 'RPS6KB2', 'PRKCE'], [])
4: Complex(['GIT1', 'MAP2K1'], [])
5: Complex(['MAP2K1', 'RAF1'], [])
6: Phosphorylation(MAP2K1, MAPK3, Phosphorylation, 202, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 8626767;93330262;15466476;11971971)])
7: Phosphorylation(RAF1, MAP2K1, Phosphorylation, None, [Evidence(bel, http://bio2rdf.org/pubmed:12551925, [], Protein kinase C (PKC) regulates activation of the Raf-1 signaling cascade by growth factors, but the mechanism by which this occurs has not been elucidated. Here we report that one mechanism involves dissociation of Raf kinase inhibitory protein (RKIP) from Raf-1. Classic and atypical but not novel PKC isoforms phosphorylate RKIP at serine 153 (Ser-153). RKIP Ser-153 phosphorylation by PKC either in vitro or in response to 12-O-tetradecanoylphorbol-13-acetate or epidermal growth factor causes release of RKIP from Raf-1)])
8: Phosphorylation(RAF1, MAP2K1, Phosphorylation, None, [Evidence(bel, http://bio2rdf.org/pubmed:12417015, [], acti-Fig. 3. PKCb in insulin signal-ing. Insulin-stimulated insulin receptor (IR) phosphorylates IRS1. IRS1 recruit various signaling molecules through SH2-phos-photyrosine interactions. IRS1-binding proteins containing an SH2 domain include Grb2, PI3K, SHP-2, and Nck. Insulin-induced ERK activation in L6 myocytes occurs largely through Raf-1 rather than Ras activation. In contrast, EGF-induced ERK acti-vation is dependent on Grb2/Sos-mediated Ras activation.)])
9: Phosphorylation(MAP2K1, MAPK3, Phosphorylation, 204, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 7867743;8626767;16464493;15466476;11971971;17081983)])
10: Phosphorylation(P2_FAMILY, MAP2K1, Phosphorylation, 25, [Evidence(bel, http://bio2rdf.org/pubmed:7873542, [], PhosphoElm data from PMID 15212693)])
11: Phosphorylation(MAP2K1, MAPK3, Phosphorylation, None, [Evidence(bel, http://bio2rdf.org/pubmed:11986622, [], in embryonic kidney cells expressing full-length DCC, NTN1 causes increased transient phosphorylation and activity of ERK1 and ERK2 but not of JNK1, JNK2 or MAPK14 this phosphorylation was mediated by MAP2K1 (aka MEK1) and/or MAP2K2 (aka MEK2))])
12: Phosphorylation(MAP2K1, MAPK1, Phosphorylation, 185, [Evidence(bel, http://bio2rdf.org/pubmed:15212693, [], 9166761;11971971;15935618)])
13: Phosphorylation(P2_FAMILY, MAP2K1, Phosphorylation, 25, [Evidence(bel, http://bio2rdf.org/pubmed:7873542, [], PhosphoElm data from PMID 15212693)])
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-2-a69375dadcfa> in <module>()
----> 1 bp = ba.process_ndex_neighborhood(['BRAF', 'MAP2K1'])

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/bel_api.pyc in process_ndex_neighborhood(gene_names, rdf_out)
     13     with open(rdf_out, 'wt') as fh:
     14         fh.write(rdf.encode('utf-8'))
---> 15     bp = process_belrdf(rdf)
     16     bp.print_statements()
     17     return bp

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/bel_api.pyc in process_belrdf(rdf_str)
     35     bp.print_statement_coverage()
     36     print "\n--- Converted BelPy Statements -------------"
---> 37     bp.print_statements()
     38     return bp
     39 

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/bel/processor.pyc in print_statements(self)
    662     def print_statements(self):
    663         for i, stmt in enumerate(self.statements):
--> 664             print "%s: %s" % (i, stmt)

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/statements.py in __str__(self)
    217         s = ("%s(%s, %s, %s, %s, %s)" %
    218                   (type(self).__name__, self.enz.name, self.sub.name, self.mod,
--> 219                    self.mod_pos, self.evidence))
    220         return s
    221 

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/statements.py in __repr__(self)
    136 
    137     def __repr__(self):
--> 138         return self.__str__()
    139 
    140 

/Users/johnbachman/Dropbox/1johndata/Knowledge File/Biology/Research/Big Mechanism/indra/indra/statements.py in __str__(self)
    132         ev_str = 'Evidence(%s, %s, %s, %s)' % \
    133                  (self.source_api, self.pmid, self.annotations,
--> 134                   print_text)
    135         return ev_str
    136 

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 74: ordinal not in range(128)

In [3]: 
```

**Comment by bgyori** [comment]
I don't get this. Are you on the preassembler branch?

**Comment by bgyori** [comment]
Actually, I get this error with ~50% probability when running the query repeatedly. I'm trying to find out what the difference is between calls.

**Comment by johnbachman** [comment]
@bgyori closed by aac35d07db72
